braindecode/augmentation/functional.py

```diff
@@ -167,22 +177,27 @@ def _ft_surrogate(x=None, f=None, eps=1, random_state=None):
     n = f.shape[-1]
     random_phase = _new_random_fft_phase[n % 2](
         f.shape[0],
+        f.shape[-2] if channel_indep else 1,
         n,
         device=device,
         random_state=random_state
     )
-    random_phase = torch.permute(
-        torch.tile(random_phase, (f.shape[-2], 1, 1)),
-        (1, 0, 2)
-    )
+    if not channel_indep:
+        random_phase = torch.tile(random_phase, (1, f.shape[-2], 1))
     if isinstance(eps, torch.Tensor):
         eps = eps.to(device)
     f_shifted = f * torch.exp(eps * random_phase)
     shifted = ifft(f_shifted, dim=-1)
     return shifted.real.float()
 
 
-def ft_surrogate(X, y, phase_noise_magnitude, random_state=None):
+def ft_surrogate(
+    X,
+    y,
+    phase_noise_magnitude,
+    channel_indep,
+    random_state=None
+):
     """FT surrogate augmentation of a single EEG channel, as proposed in [1]_.
```

Comment:
This line should be adapted as the new implementation is for multiple channels

Reply:
This was intentional, the rationale being to set defaults at Transform level and force the user to take a decision for functionals (which are "lower-level"). The same designs is adopted for other transforms and functionals.